### ISSUE TITLE
Allow setting context via command line

### ISSUE THREAD

**Issue by RyanZim** [postcss/postcss-cli#52]
As mentioned in https://github.com/postcss/postcss-cli/issues/51.

Currently, we have `--env` to allow setting `process.env.NODE_ENV` for `postcss.config.js`. We also have command-line flags to override the syntax, parser, stringifier, & map options specified in the config.

We should have an additional command-line flag to allow passing other values via `ctx` to `postcss.config.js`.

My strawman proposal:

    postcss --ctx {doIt: true, someKey: 'somevalue'}

This needs discussion.

**Comment by RyanZim** [comment]
@michael-ciniawsky ping?

**Comment by michael-ciniawsky** [comment]
I currently expose the `config.options` to `ctx`

```bash
postcss -p sugarss -m -e production
```
```js
module.exports = ({ options, file, env }) => ({
   parser: file.extname === '.sss' ? options.parser : false
   map: options.map || { inline: false }
   plugins: {
     'postcss-import': { root: file.dirname }
     'cssnano': env === 'prodution' ? {} : false
   }
})
```  

The following is not working and is a mess, how did postcss-cli v2 `--use` work ? Someting like `options.plugins.key = value` would still be valuable though

```bash
postcss -u || --ctx  { key: value }
```

**Comment by RyanZim** [comment]
@michael-ciniawsky So what are we doing here?

**Comment by michael-ciniawsky** [comment]
Options can be set via the CLI, only plugin options must be defined in the config itself

```bash
postcss -m -p -t -s
```

**postcss.config.js**
```
module.exports = (ctx) => ({
   map: cts.options.map || fallback,
   parser: ctx.options.parser || fallback,
   syntax: ctx.options. syntax,
   stringifier: ctx.options.stringifier
  plugins: {...plugins}
})
```

**Comment by RyanZim** [comment]
So we don't need custom context?

**Comment by michael-ciniawsky** [comment]
Well It  would into the direction like `-- ctx ctx.key value` which is similiar to `-u plugin.option value` and we decided to not support atm. In the future we can try to add the `-u ..` to the context e.g `ctx.option.plugins`

```js
module.exports = (ctx) => ({
   map: cts.options.map || fallback,
   parser: ctx.options.parser || fallback,
   syntax: ctx.options. syntax,
   stringifier: ctx.options.stringifier,
   plugins: { 
     'posthtml-plugin': ctx.options.plugin
   }
})
```

**Comment by RyanZim** [comment]
> In the future we can try to add the -u .. to the context e.g ctx.option.plugins

:+1: Perhaps should be `ctx.plugins`, but, yeah.

---

Can you document what's currently available via `ctx`?

**Comment by michael-ciniawsky** [comment]
Yep `ctx.plugins`  of course 😛 . yep I will make docs about it

**Comment by michael-ciniawsky** [comment]
Fixed by #86
